src/orange/construct/OrangeInput.hh

```diff
@@ -97,13 +103,37 @@ struct UnitInput
     explicit operator bool() const { return !volumes.empty(); }
 };
 
+//---------------------------------------------------------------------------//
+/*!
+ * Input definition for a rectangular array universe.
+ */
+struct RectArrayInput
+{
+    // Grid boundaries in x, y, and z
+    Array<std::vector<double>, 3> grid;
+
+    // Daughter loading
+    std::vector<DaughterInput> daughters;
+
+    BoundingBox bbox;  //!< Outer bounding box
+
+    // Unit metadata
+    Label label;
+
+    //! Whether the universe definition is valid
+    explicit operator bool() const { return !daughters.empty(); }
```

Comment:
At the minimum we should test that the grid isn't empty (validation for monotonicity etc. would be good to add in the builder maybe?)
```suggestion
    explicit operator bool() const
    {
        return !daughters.empty()
               && std::all_of(grid.begin(), grid.end(), [](auto& v) {
                      return v.size() >= 2;
                  });
    }
```

Reply:
Looks you already committed this